docs/proposal/aws-resource-handling.md

```diff
@@ -38,7 +38,11 @@ Where possible use [client tokens](https://docs.aws.amazon.com/AWSEC2/latest/API
 
 ## Tagging of resources
 
-Resources that are managed by the controllers/actuators should be tagged with: `kubernetes.io/cluster/<name or id>=owned` and `sigs.k8s.io/cluster-api-provider-aws=true`. The latter tag being used to differentiate from resources managed by other tools/components that make use of the common tag.
+Resources handled by these components fall into one of three categories:
+
+1. Fully-managed resources whose lifecycle is tied to the cluster. These resources should be tagged with `sigs.k8s.io/cluster-api-provider/<name or id>=owned` and `sigs.k8s.io/cluster-api-provider-aws/managed=true` and the actuator is expected to keep these resources as closely in sync with the spec as possible.
+2. Resources whose management is shared with the in-cluster aws cloud provider,, such as a security group for load balancer ingress rules. These resources should be tagged with `sigs.k8s.io/cluster-api-provider/<name or id>=owned` and `kubernetes.io/cluster/<name or id>=owned`, with the latter being the tag defined by the cloud provider. These resources are create/delete only: that is to say their ongoing management is "handed off" to the cloud provider.
```

Comment:
We've found two:

1. Persistent volume support via EBS (requires tagging the instance)
2. Security group rules for ELBs (requires tagging the security group)

This PR covers those cases by adding the "cluster-owned" tag back, but I expect there will be more cases that this change misses.

Reply:
Ah, gotcha. In that case, would it make more sense to apply `kubernetes.io/cluster/<name or id>=shared` instead? Since the cloud provider would not actually own the resource? It should still be able to query and make use of the resource.

Reply:
Hmm, maybe? The comment in the cloud provider gives me pause: https://github.com/kubernetes/kubernetes/blob/d7103187a37dcfff79077c80a151e98571487628/pkg/cloudprovider/providers/aws/tags.go#L45-L52

>    // ResourceLifecycleOwned is the value we use when tagging resources to indicate
>    // that the resource is considered owned and managed by the cluster,
>    // and in particular that the lifecycle is tied to the lifecycle of the cluster.
> ...
>	// ResourceLifecycleShared is the value we use when tagging resources to indicate
>	// that the resource is shared between multiple clusters, and should not be destroyed
>	// if the cluster is destroyed.

And at least as far as I can see, the provider never reads the value of the tag (it only checks for its existence: https://github.com/kubernetes/kubernetes/blob/d7103187a37dcfff79077c80a151e98571487628/pkg/cloudprovider/providers/aws/tags.go#L133 ).

Maybe the thing to do is tag them with a new value, like `provided`, and PR the cloud provider to adopt those semantics (since it'd just be a `const` and a comment at this point)?